src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -234,6 +234,11 @@ protected function finalizeValue($value)
         }
 
         foreach ($this->children as $name => $child) {
+            if ($child->isDeprecated()) {
+                $msg = sprintf('The child node "%s" at path "%s" is deprecated.', $name, $this->getPath());
```

Comment:
Yes, I’m agree  with you. But I did’t find a good way to do this.

Maybe by passing an optionnal argument to `isDeprecated()` method?